cmd/nvidia-ctk-installer/container/runtime/crio/config_test.go

```diff
@@ -0,0 +1,606 @@
+		},
+		{
+			description: "hook mode: existing hook file",
+			containerOptions: container.Options{
+				RuntimeName: "nvidia",
+				RuntimeDir:  "/usr/bin",
+				RestartMode: "none",
+			},
+			options: Options{
+				configMode:   "hook",
+				hooksDir:     "{{ .testRoot }}/etc/crio/hooks.d",
+				hookFilename: "99-nvidia.json",
+			},
+			prepareEnvironment: func(t *testing.T, testRoot string) error {
+				hooksDir := filepath.Join(testRoot, "etc/crio/hooks.d")
+				require.NoError(t, os.MkdirAll(hooksDir, 0755))
+
+				// Create existing hook with old path
+				existingHookJSON := `{
+  "version": "1.0.0",
+  "hook": {
+    "path": "/old/path/nvidia-container-runtime-hook",
+    "args": [
+      "nvidia-container-runtime-hook",
+      "prestart"
+    ]
+  },
+  "when": {
+    "always": true
+  },
+  "stages": [
+    "prestart"
+  ]
+}`
+
+				hookPath := filepath.Join(hooksDir, "99-nvidia.json")
+				err := os.WriteFile(hookPath, []byte(existingHookJSON), 0600)
+				require.NoError(t, err)
+
+				return nil
+			},
+			assertSetupPostConditions: func(t *testing.T, testRoot string) error {
+				hookPath := filepath.Join(testRoot, "etc/crio/hooks.d/99-nvidia.json")
+				require.FileExists(t, hookPath)
+
+				actual, err := os.ReadFile(hookPath)
+				require.NoError(t, err)
+
+				expectedBinary := filepath.Join("/usr/bin", config.NVIDIAContainerRuntimeHookExecutable)
+				expected := fmt.Sprintf(`{
+  "version": "1.0.0",
+  "hook": {
+    "path": "%s",
+    "args": ["nvidia-container-runtime-hook", "prestart"],
+    "env": [
+      "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+    ]
+  },
+  "when": {
+    "always": true,
+    "commands": [".*"]
+  },
+  "stages": ["prestart"]
+}`, expectedBinary)
+
+				// Parse both JSONs to compare them semantically (ignoring formatting differences)
+				var expectedJSON, actualJSON interface{}
+				require.NoError(t, json.Unmarshal([]byte(expected), &expectedJSON))
+				require.NoError(t, json.Unmarshal(actual, &actualJSON))
+				require.Equal(t, expectedJSON, actualJSON)
+
+				return nil
+			},
+			assertCleanupPostConditions: func(t *testing.T, testRoot string) error {
+				hookPath := filepath.Join(testRoot, "etc/crio/hooks.d/99-nvidia.json")
+				require.NoFileExists(t, hookPath)
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			// Update any paths as required
+			testRoot := t.TempDir()
+			tc.containerOptions.Config = strings.ReplaceAll(tc.containerOptions.Config, "{{ .testRoot }}", testRoot)
+			tc.options.hooksDir = strings.ReplaceAll(tc.options.hooksDir, "{{ .testRoot }}", testRoot)
+
+			// Prepare the test environment
+			if tc.prepareEnvironment != nil {
+				require.NoError(t, tc.prepareEnvironment(t, testRoot))
+			} else if tc.options.configMode == "hook" {
+				// Ensure hooks directory exists for hook mode
+				hooksDir := tc.options.hooksDir
+				require.NoError(t, os.MkdirAll(hooksDir, 0755))
+			}
```

Comment:
Why is this not just implemented in `prepareEnvironment`? Also, we explicitly create the hook dir as part of the installation, so this should not be reqiured in any case.